src/components/common/buttons/ButtonWithDoneState.js

```diff
@@ -0,0 +1,62 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { StyleSheet, View } from 'react-native'
+import Icon from '../view/Icon'
+import CustomButton from './CustomButton'
+
+const NOT_EXECUTED = 'NOT_EXECUTED'
+const EXECUTED = 'EXECUTED'
+const DONE = 'DONE'
+const TRANSITION_TIME = 1000
+
+const styles = StyleSheet.create({
+  iconButtonWrapper: {
+    minHeight: 28,
+    justifyContent: 'center',
+  },
+})
+
+const ButtonWithDoneState = ({ toCopy, children, onPress, onPressDone, iconColor, withoutDone, ...props }) => {
+  const mode = props.mode || 'contained'
+  const [state, setState] = useState(NOT_EXECUTED)
```

Comment:
isExecuted / setExecuted

pls avoid setState as a) it's correlates with class component method b) state is something complex, it's associated with full component state object while here you're describing just one variable - 'is transaction executed' flag

Reply:
Done. renamed